### PR TITLE
Component breakeup

### PR DESCRIPTION
**Pull request by JanekJaniak**

### CODE REVIEW

File: src/components/IngredientsTable.vue [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+</template>  
+
+<script>
+  export default {
+    name: 'IngredientsTable',
+      data() {
+        return {
+          value: ''
+        }
+      },
```
Comment (jan-gajdowicz) [inline src/components/IngredientsTable.vue]: wrong indent

File: src/components/IngredientsTable.vue [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    
+    props: {
+      ingredients: Array,
+      vatOptions: Array,
+      tHeader: Array
+    },
+
+    methods: {
+      handleInput(event, index, field) {
+        this.$emit('inputChange',{value:event.target.value, index, field})
```
Comment (jan-gajdowicz) [inline src/components/IngredientsTable.vue]: spacing

File: src/components/IngredientsTable.vue [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      },
+
+      handleSelect(event, index) {
+        this.$emit('vatChange',{value:event.target.value, index})
```
Comment (jan-gajdowicz) [inline src/components/IngredientsTable.vue]: spacing

File: src/components/IngredientsTable.vue [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        return netValue;
+      },
+
+      calculateGrossValue(quantity, price, vat, index) {
+        const grossValue = (quantity * price * vat).toFixed(2);
+
+        this.$emit('changeGrossValue', {value:grossValue, index})
+
+        return grossValue;
+      },
```
Comment (jan-gajdowicz) [inline src/components/IngredientsTable.vue]: this should be a single function

File: src/components/MealCostCalculator.vue [hunk 5]
```diff
@@ -192,89 +142,89 @@ export default {
...
+      this.meal.ingredients[index].vat = value
+    },
+
+    updateNetValue ({value, index}) {
+      this.meal.ingredients[index].net = value
+    },
+
+    updateGrossValue ({value, index}) {
+      this.meal.ingredients[index].gross = value
+    },
```
Comment (jan-gajdowicz) [inline src/components/MealCostCalculator.vue]: This should be a single function

File: src/components/MealCostCalculator.vue [hunk 6]
```diff
@@ -192,89 +142,89 @@ export default {
...
+
+    removeItem(id) {
+      this.meal.ingredients = this.meal.ingredients.filter(ingredient =>ingredient.id != id)
```
Comment (jan-gajdowicz) [inline src/components/MealCostCalculator.vue]: spacing

File: src/style.sass [hunk 7]
```diff
@@ -1,4 +1,4 @@
 @import './styles/button.sass'
 @import './styles/form.sass'
 @import './styles/table.sass'
-@import './styles/typo.sass'
\ No newline at end of file
+@import './styles/typo.sass'
```
Comment (jan-gajdowicz) [inline src/style.sass:4]: whitespace

File: src/styles/table.sass [hunk 8]
```diff
@@ -19,5 +19,3 @@
 
     &-no_border
       border: none
-
-
```
Comment (jan-gajdowicz) [inline src/styles/table.sass:23]: whitespace

File: src/components/MealCostCalculator.vue [hunk 9]
```diff
@@ -4,68 +4,17 @@
...
-
-      button.button.button-dim(@click="addIngredient()") Add new ingredient
+    ingredients-table(
+      :ingredients="meal.ingredients"
+      :vatOptions="vatOptions",
+      :tHeader="tHeader"
+      @inputChange="updateInputValue"
+      @vatChange="updateVat"
+      @changeNetValue="updateNetValue"
+      @changeGrossValue="updateGrossValue"
```
Comment (jan-gajdowicz) [inline src/components/MealCostCalculator.vue]: `@update="updateValue"` - single event and function

File: src/components/MealCostCalculator.vue [hunk 10]
```diff
@@ -4,68 +4,17 @@
...
+      @removeAction="removeItem"
```
Comment (jan-gajdowicz) [inline src/components/MealCostCalculator.vue]: @remove

File: src/components/MealCostCalculator.vue [hunk 11]
```diff
@@ -4,68 +4,17 @@
...
+      @newItem="addIngredient"
```
Comment (jan-gajdowicz) [inline src/components/MealCostCalculator.vue]: @create

File: src/components/IngredientsTable.vue [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+    methods: {
+      handleInput(event, index, field) {
+        this.$emit('inputChange',{value:event.target.value, index, field})
+      },
+
+      handleSelect(event, index) {
+        this.$emit('vatChange',{value:event.target.value, index})
+      },
+
```
Comment (jan-gajdowicz) [inline src/components/IngredientsTable.vue]: this should be a single function

**Review by jan-gajdowicz — COMMENTED** [review]
Change PR name to `Component breakup`

File: src/components/IngredientsTable.vue (hunk 1 above)
Comment (JanekJaniak) [reply src/components/IngredientsTable.vue]: :heavy_check_mark:

File: src/components/IngredientsTable.vue (hunk 2 above)
Comment (JanekJaniak) [reply src/components/IngredientsTable.vue]: :heavy_check_mark:

File: src/components/IngredientsTable.vue (hunk 3 above)
Comment (JanekJaniak) [reply src/components/IngredientsTable.vue]: :heavy_check_mark:

File: src/styles/table.sass (hunk 8 above)
Comment (JanekJaniak) [reply src/styles/table.sass:23]: :heavy_check_mark:

File: src/style.sass (hunk 7 above)
Comment (JanekJaniak) [reply src/style.sass:4]: :heavy_check_mark:

File: src/components/MealCostCalculator.vue (hunk 10 above)
Comment (JanekJaniak) [reply src/components/MealCostCalculator.vue]: :heavy_check_mark:

File: src/components/IngredientsTable.vue (hunk 4 above)
Comment (JanekJaniak) [reply src/components/IngredientsTable.vue]: :heavy_check_mark:

File: src/components/MealCostCalculator.vue (hunk 5 above)
Comment (JanekJaniak) [reply src/components/MealCostCalculator.vue]: :heavy_check_mark:

File: src/components/MealCostCalculator.vue (hunk 9 above)
Comment (JanekJaniak) [reply src/components/MealCostCalculator.vue]: :heavy_check_mark:

File: src/components/MealCostCalculator.vue (hunk 11 above)
Comment (JanekJaniak) [reply src/components/MealCostCalculator.vue]: :heavy_check_mark:

File: src/components/IngredientsTable.vue (hunk 12 above)
Comment (JanekJaniak) [reply src/components/IngredientsTable.vue]: :heavy_check_mark:

**Review by jan-gajdowicz — APPROVED** [review]

**Pull request merged**